raiden/transfer/channel.py

```diff
@@ -844,6 +844,29 @@ def get_amount_locked(end_state: NettingChannelEndState) -> TokenAmount:
     return total_pending + total_unclaimed + total_unclaimed_onchain
 
 
+def get_batch_unlock_gain(
+        channel_state: NettingChannelState,
+) -> Tuple[TokenAmount, TokenAmount]:
+    """Collect amounts for unlocked/unclaimed locks and onchain unlocked locks.
+    Note: this function does not check expiry, so the values make only sense during settlement.
+
+    Returns:
+        gain_from_partner_locks: locks amount received and unlocked on-chain
+        from_from_our_locks: locks amount which are unlocked or unclaimed
+    """
+    gain_from_partner_locks = sum(
+        unlock.lock.amount
+        for unlock in channel_state.partner_state.secrethashes_to_onchain_unlockedlocks.values()
+    )
+    gain_from_our_locks = sum(
+        lock.amount
+        for lock in channel_state.our_state.secrethashes_to_lockedlocks.values()
+    ) + sum(
+        lock.amount for lock in channel_state.our_state.secrethashes_to_unlockedlocks.values()
```

Comment:
This corner case is a surprise to people, could you please add some explanation to it?

Reply:
Added some docstring explaning why we collect this value.